apps/reaper/lib/reaper/event/event_handler.ex

```diff
@@ -72,6 +85,34 @@ defmodule Reaper.Event.EventHandler do
     Extractions.update_last_fetched_timestamp(ingestion_id)
   end
 
+  def handle_event(%Brook.Event{
+        type: dataset_delete(),
+        data: %Dataset{} = dataset
+      }) do
+    dataset_delete()
+    |> add_event_count(dataset.id)
+
+    {:ok, extractions} = Brook.ViewState.get_all(@instance_name, :extractions)
+
+    extractions_to_delete =
+      Enum.filter(extractions, fn {key, e} ->
+        with {:ok, ingestion} <- Map.fetch(e, "ingestion") do
+          ingestion[:targetDataset] == dataset[:id]
+        else
+          :error -> Logger.error("Extraction #{key} does not have an Ingestion object")
```

Comment:
As of today, every extraction has an ingestion object. Wanted to just add this error messaging in case that changes someday.